File: app/src/main/java/com/jzoft/ygohelper/utils/HttpCaller.java
package com.jzoft.ygohelper.utils;

/**
 * Created by jjimenez on 13/10/16.
 */
public interface HttpCaller {

    byte[] getCall(String url) throws NotFound;


    class NotFound extends Exception {
        private String url;

        public NotFound(String url) {
            this.url = url;
        }

        public String getUrl() {
            return url;
        }
    }
}

File: app/src/main/java/com/jzoft/ygohelper/utils/HttpCaller.kt
package com.jzoft.ygohelper.utils

/**
 * Created by jjimenez on 13/10/16.
 */
interface HttpCaller {

    @Throws(HttpCaller.NotFound::class)
    fun getCall(url: String): ByteArray


    class NotFound(val url: String) : Exception()
}
